packages/sanity/src/structure/panes/document/statusBar/DocumentStatusBarActions.tsx

```diff
@@ -22,9 +24,21 @@ interface DocumentStatusBarActionsInnerProps {
 
 function DocumentStatusBarActionsInner(props: DocumentStatusBarActionsInnerProps) {
   const {disabled, showMenu, states} = props
-  const {__internal_tasks} = useDocumentPane()
+  const {__internal_tasks, schemaType, openPath} = useDocumentPane()
   const [firstActionState, ...menuActionStates] = states
   const [buttonElement, setButtonElement] = useState<HTMLButtonElement | null>(null)
+  const {enabled: isTreeArrayEditingEnabled} = useTreeArrayEditingEnabled()
+
+  const isDialogOpen = useMemo(
+    () => shouldArrayDialogOpen(schemaType, openPath),
```

Comment:
For the use of this function this feels like the function could be renamed to `isArrayDialogSupported` or , `shouldUseArrayDialog` instead of `shouldArrayDialogOpen`
Also, given treeEditing could be disabled, I suggest we first check for that and later for the function.
```suggestion
  const isTreeArrayEditingEnabledOpen = useMemo(
    () => isTreeArrayEditingEnabled && shouldArrayDialogOpen(schemaType, openPath),
    [openPath, schemaType, isTreeArrayEditingEnabled],
  )

```

Reply:
Fair on the suggestion!
I think the naming should remain the same though, the shouldArrayDialogOpen doesn't only check if it's supported, it literally checks if the dialog should be allowed open based on where it is (path and the schematype). The dialog can be "supported" and yet not be open in a specific scenario :)